tests/unit/amo/components/TestHeader.js

```diff
@@ -40,4 +63,95 @@ describe(__filename, () => {
     expect(root.find('.Header-title').prop('children'))
       .toContain('Firefox Add-ons');
   });
+
+  it('displays `login` text when user is not connected', () => {
+    const { store } = dispatchClientMetadata();
+    const fakeI18n = getFakeI18nInst();
+    const allProps = { i18n: fakeI18n, location: {}, query: '' };
+
+    const wrapper = mount(
+      <Provider store={store}>
+        <I18nProvider i18n={fakeI18n}>
+          <Header {...allProps} />
+        </I18nProvider>
+      </Provider>
+    );
+
+    expect(wrapper.find(AuthenticateButton).text())
+      .toEqual('Register or Log in');
+    expect(wrapper.find(DropdownMenu)).toHaveLength(0);
+  });
+
+  it('displays a menu and the username when user is logged in', () => {
+    const { store } = dispatchSignInActions({ username: 'babar' });
+    const fakeI18n = getFakeI18nInst();
+    const allProps = { i18n: fakeI18n, location: {}, query: '' };
+
+    const wrapper = mount(
+      <Provider store={store}>
+        <I18nProvider i18n={fakeI18n}>
+          <Header {...allProps} />
+        </I18nProvider>
+      </Provider>
+    );
+
+    expect(wrapper.find(DropdownMenu)).toHaveLength(1);
+    expect(wrapper.find(DropdownMenu)).toIncludeText('babar');
+  });
+
+  it('allows to log out a logged user', () => {
+    const { store } = dispatchSignInActions({ username: 'babar' });
+    const fakeI18n = getFakeI18nInst();
+    const allProps = { i18n: fakeI18n, location: {}, query: '' };
+    const mockApi = sinon.mock(api);
+
+    mockApi
+      .expects('logOutFromServer')
+      .once()
+      .returns(Promise.resolve());
+
+    const wrapper = mount(
+      <Provider store={store}>
+        <I18nProvider i18n={fakeI18n}>
+          <Header {...allProps} />
+        </I18nProvider>
+      </Provider>
+    );
+
+    wrapper.find(DropdownMenuItem).find('button').simulate('click');
+
+    mockApi.verify();
+  });
+
+  describe('mapStateToProps', () => {
```

Comment:
Do we need these tests here? @kumar303 has pointed out we can test `mapStateToProps` are working by testing with dispatch and looking at the output of the rendered component itself based on what `mapStateToProps` sends. I suppose testing `mapStateToProps` is more direct unit testing but it feels more removed.